vm/src/builtins/singletons.rs

```diff
@@ -39,7 +41,7 @@ impl Constructor for PyNone {
     }
 }
 
-#[pyimpl(with(Constructor))]
+#[pyimpl(flags(BASETYPE), with(Constructor, AsNumber))]
```

Comment:
```suggestion
#[pyimpl(with(Constructor, AsNumber))]
```
It seems PyNone does not have `BASETYPE` flags in cpython. Is there any reason for adding it?